File: SmartTriage-server/src/main/java/com/smartTriage/smartTriage_server/module/clinicalsigns/service/ClinicalSignDefinitions.java
package com.smartTriage.smartTriage_server.module.clinicalsigns.service;

import com.smartTriage.smartTriage_server.module.clinicalsigns.entity.ClinicalSignCategory;
import com.smartTriage.smartTriage_server.module.triage.entity.TriageRecord;

import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.function.Predicate;

/**
 * Canonical catalog of the 54 clinical signs we track over time.
 *
 * Single source of truth on the server for:
 *   1. Mapping each sign_code to its ClinicalSignCategory (so the request
 *      DTO doesn't have to carry category — the server derives it)
 *   2. Bootstrapping the timeline from a fresh TriageRecord — we know how
 *      to read every triage flag and which sign_code it corresponds to
 *
 * The frontend has its own parallel catalog (`clinicalSignDefinitions.ts`)
 * with the same codes plus human-readable labels. Drift between the two
 * would mean the doctor sees "Update sign" for a code the server doesn't
 * recognize — keep them in sync when adding new signs.
 *
 * The V27 migration's backfill SQL also encodes this mapping; it's
 * intentionally duplicated there because Flyway migrations run before
 * the application boots.
 */
public final class ClinicalSignDefinitions {

    private ClinicalSignDefinitions() {}

    /**
     * Each entry: (sign_code, category, "is positive on this triage record"
     * predicate, optional getter for an associated numeric value such as
     * glucose). The numeric getter returns null when the sign carries no
     * numeric value or the field is unset.
     */
    public record SignMapping(
            String code,
            ClinicalSignCategory category,
            Predicate<TriageRecord> isPositive,
            Function<TriageRecord, Double> numericValue
    ) {}

    private static SignMapping flag(String code, ClinicalSignCategory cat, Predicate<TriageRecord> p) {
        return new SignMapping(code, cat, p, t -> null);
    }

    private static SignMapping flagWithNumeric(
            String code, ClinicalSignCategory cat,
            Predicate<TriageRecord> p, Function<TriageRecord, Double> num) {
        return new SignMapping(code, cat, p, num);
    }

    /**
     * Full catalog. Order is the order signs appear on the Rwanda triage
     * form (Section 1, 1b, 3, 4, special) — preserved here so the doctor's
     * timeline reads in form-order rather than alphabetical.
     */
    public static final List<SignMapping> ALL = List.of(
            // ── Emergency (Section 1) ──
            flag("EMERGENCY_AIRWAY_COMPROMISE",          ClinicalSignCategory.EMERGENCY, TriageRecord::isHasAirwayCompromise),
            flag("EMERGENCY_BREATHING_DISTRESS",         ClinicalSignCategory.EMERGENCY, TriageRecord::isHasBreathingDistress),
            flag("EMERGENCY_SEVERE_RESPIRATORY_DISTRESS",ClinicalSignCategory.EMERGENCY, TriageRecord::isHasSevereRespiratoryDistress),
            flag("EMERGENCY_CARDIAC_ARREST",             ClinicalSignCategory.EMERGENCY, TriageRecord::isHasCardiacArrest),
            flag("EMERGENCY_UNCONTROLLED_HAEMORRHAGE",   ClinicalSignCategory.EMERGENCY, TriageRecord::isHasUncontrolledHaemorrhage),
            flag("EMERGENCY_STAB_GUN_WOUND_NECK_CHEST",  ClinicalSignCategory.EMERGENCY, TriageRecord::isHasStabGunWoundNeckChest),
            flagWithNumeric("EMERGENCY_CONVULSIONS",     ClinicalSignCategory.EMERGENCY, TriageRecord::isHasConvulsions, TriageRecord::getConvulsionGlucose),
            flagWithNumeric("EMERGENCY_COMA",            ClinicalSignCategory.EMERGENCY, TriageRecord::isHasComa, TriageRecord::getComaGlucose),
            flag("EMERGENCY_HYPOGLYCAEMIA",              ClinicalSignCategory.EMERGENCY, TriageRecord::isHasHypoglycaemia),
            flag("EMERGENCY_PURPURIC_RASH",              ClinicalSignCategory.EMERGENCY, TriageRecord::isHasPurpuricRash),
            flag("EMERGENCY_BURN_FACE_INHALATION",       ClinicalSignCategory.EMERGENCY, TriageRecord::isHasBurnFaceInhalation),

            // ── Pediatric Emergency (Section 1b) ──
            flag("PEDS_EMERGENCY_CENTRAL_CYANOSIS",          ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildCentralCyanosis),
            flag("PEDS_EMERGENCY_PULSE_LOW_OR_ABSENT",       ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildPulseLowOrAbsent),
            flag("PEDS_EMERGENCY_COLD_HANDS_COMPOSITE",      ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildColdHandsComposite),
            flag("PEDS_EMERGENCY_COLD_HANDS_LETHARGIC",      ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildColdHandsLethargic),
            flag("PEDS_EMERGENCY_COLD_HANDS_PULSE_WEAK_FAST",ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildColdHandsPulseWeakFast),
            flag("PEDS_EMERGENCY_COLD_HANDS_CAP_REFILL",     ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildColdHandsCapRefill),
            flag("PEDS_EMERGENCY_SEVERE_DEHYDRATION",        ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildSevereDehydration),
            flag("PEDS_EMERGENCY_DEHYDRATION_SKIN_PINCH",    ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildDehydrationSkinPinch),
            flag("PEDS_EMERGENCY_DEHYDRATION_LETHARGY",      ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildDehydrationLethargy),
            flag("PEDS_EMERGENCY_DEHYDRATION_SUNKEN_EYES",   ClinicalSignCategory.PEDIATRIC_EMERGENCY, TriageRecord::isChildDehydrationSunkenEyes),

            // ── mSAT Very Urgent (Section 3) ──
            flag("MSAT_VU_FOCAL_NEUROLOGIC_DEFICIT",     ClinicalSignCategory.MSAT_VU, TriageRecord::isVuFocalNeurologicDeficit),
            flagWithNumeric("MSAT_VU_ALTERED_MENTAL_STATUS", ClinicalSignCategory.MSAT_VU, TriageRecord::isVuAlteredMentalStatus, TriageRecord::getVuNeurologicalGlucose),
            flag("MSAT_VU_CHEST_PAIN",                   ClinicalSignCategory.MSAT_VU, TriageRecord::isVuChestPain),
            flag("MSAT_VU_POISONING_OVERDOSE",           ClinicalSignCategory.MSAT_VU, TriageRecord::isVuPoisoningOverdose),
            flag("MSAT_VU_PREGNANT_ABDOMINAL_PAIN",      ClinicalSignCategory.MSAT_VU, TriageRecord::isVuPregnantAbdominalPain),
            flag("MSAT_VU_COUGHING_VOMITING_BLOOD",      ClinicalSignCategory.MSAT_VU, TriageRecord::isVuCoughingVomitingBlood),
            flagWithNumeric("MSAT_VU_DIABETIC_HIGH_GLUCOSE", ClinicalSignCategory.MSAT_VU, TriageRecord::isVuDiabeticHighGlucose, TriageRecord::getVuDiabeticGlucose),
            flag("MSAT_VU_AGGRESSION",                   ClinicalSignCategory.MSAT_VU, TriageRecord::isVuAggression),
            flag("MSAT_VU_SHORTNESS_OF_BREATH",          ClinicalSignCategory.MSAT_VU, TriageRecord::isVuShortnessOfBreath),
            flag("MSAT_VU_BURN_OVER_20_PERCENT",         ClinicalSignCategory.MSAT_VU, TriageRecord::isVuBurnOver20Percent),
            flag("MSAT_VU_OPEN_FRACTURE",                ClinicalSignCategory.MSAT_VU, TriageRecord::isVuOpenFracture),
            flag("MSAT_VU_THREATENED_LIMB",              ClinicalSignCategory.MSAT_VU, TriageRecord::isVuThreatenedLimb),
            flag("MSAT_VU_EYE_INJURY",                   ClinicalSignCategory.MSAT_VU, TriageRecord::isVuEyeInjury),
            flag("MSAT_VU_LARGE_JOINT_DISLOCATION",      ClinicalSignCategory.MSAT_VU, TriageRecord::isVuLargeJointDislocation),
            flag("MSAT_VU_SEVERE_MECHANISM_OF_INJURY",   ClinicalSignCategory.MSAT_VU, TriageRecord::isVuSevereMechanismOfInjury),
            flag("MSAT_VU_VERY_SEVERE_PAIN",             ClinicalSignCategory.MSAT_VU, TriageRecord::isVuVerySeverePain),
            flag("MSAT_VU_PREGNANT_ABDOMINAL_TRAUMA",    ClinicalSignCategory.MSAT_VU, TriageRecord::isVuPregnantAbdominalTrauma),

            // ── mSAT Urgent (Section 4) ──
            flag("MSAT_URG_UNABLE_TO_DRINK_VOMITS",      ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgUnableToDrinkVomits),
            flag("MSAT_URG_ABDOMINAL_PAIN",              ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgAbdominalPain),
            flag("MSAT_URG_VERY_PALE",                   ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgVeryPale),
            flag("MSAT_URG_PREGNANT_VAGINAL_BLEEDING",   ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgPregnantVaginalBleeding),
            flagWithNumeric("MSAT_URG_DIABETIC_VERY_HIGH_GLUCOSE", ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgDiabeticVeryHighGlucose, TriageRecord::getUrgDiabeticGlucose),
            flag("MSAT_URG_FINGER_TOE_DISLOCATION",      ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgFingerToeDislocation),
            flag("MSAT_URG_CLOSED_FRACTURE",             ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgClosedFracture),
            flag("MSAT_URG_BURN_WITHOUT_URGENT_SIGNS",   ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgBurnWithoutUrgentSigns),
            flag("MSAT_URG_PREGNANT_TRAUMA_NON_ABDOMINAL", ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgPregnantTraumaNonAbdominal),
            flag("MSAT_URG_MODERATE_PAIN",               ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgModeratePain),
            flag("MSAT_URG_LACERATION_ABSCESS",          ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgLacerationAbscess),
            flag("MSAT_URG_FOREIGN_BODY_ASPIRATION",     ClinicalSignCategory.MSAT_URG, TriageRecord::isUrgForeignBodyAspiration),

            // ── Special considerations (form footer) ──
            flag("SPECIAL_ACUTE_TRAUMA",       ClinicalSignCategory.SPECIAL, TriageRecord::isSpecialAcuteTrauma),
            flag("SPECIAL_SEIZURE_HISTORY",    ClinicalSignCategory.SPECIAL, TriageRecord::isSpecialSeizureHistory),
            flag("SPECIAL_ASSAULT_ABUSE",      ClinicalSignCategory.SPECIAL, TriageRecord::isSpecialAssaultAbuse),
            flag("SPECIAL_SUICIDE_ATTEMPT",    ClinicalSignCategory.SPECIAL, TriageRecord::isSpecialSuicideAttempt)
    );

    /** Quick lookup: sign_code → category. Used by the request handler to
     *  derive category server-side rather than trusting the client. */
    public static final Map<String, ClinicalSignCategory> CATEGORY_BY_CODE =
            ALL.stream().collect(java.util.stream.Collectors.toUnmodifiableMap(
                    SignMapping::code, SignMapping::category));

    /** True when `code` is a recognised clinical sign code. */
    public static boolean isKnownCode(String code) {
        return code != null && CATEGORY_BY_CODE.containsKey(code);
    }
}

File: SmartTriage_Frontend_V6/src/modules/visit/clinicalSignDefinitions.ts
/* ═══════════════════════════════════════════════════════════════
   Frontend catalog of the 54 clinical signs we track over time.

   Mirrors the backend `ClinicalSignDefinitions.java` — codes and
   categories MUST stay in sync. Adding a new sign requires updating:
     1. backend ClinicalSignDefinitions.java (mapping + category)
     2. backend V27 (or successor) migration's backfill INSERT
     3. this file (label + category)
   The server rejects unknown codes, so frontend-only additions silently
   fail validation rather than producing bad data.

   Order matches the Rwanda triage form order so the UI reads in form
   order rather than alphabetical.
   ═══════════════════════════════════════════════════════════════ */

import type { ClinicalSignCategory } from '@/api/clinicalSigns';

export interface ClinicalSignDefinition {
  code: string;
  label: string;
  category: ClinicalSignCategory;
  /** True when this sign carries an associated glucose value. */
  carriesNumeric: boolean;
  /** Helper hint for the Update form's numeric field. */
  numericLabel?: string;
}

export const CLINICAL_SIGN_DEFINITIONS: ClinicalSignDefinition[] = [
  // ── Emergency (Section 1) ──
  { code: 'EMERGENCY_AIRWAY_COMPROMISE',           label: 'Airway compromise',                category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_BREATHING_DISTRESS',          label: 'Breathing distress',               category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_SEVERE_RESPIRATORY_DISTRESS', label: 'Severe respiratory distress',      category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_CARDIAC_ARREST',              label: 'Cardiac arrest',                   category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_UNCONTROLLED_HAEMORRHAGE',    label: 'Uncontrolled haemorrhage',         category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_STAB_GUN_WOUND_NECK_CHEST',   label: 'Stab / gunshot wound to neck or chest', category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_CONVULSIONS',                 label: 'Convulsions',                      category: 'EMERGENCY', carriesNumeric: true,  numericLabel: 'Glucose (mmol/L)' },
  { code: 'EMERGENCY_COMA',                        label: 'Coma',                             category: 'EMERGENCY', carriesNumeric: true,  numericLabel: 'Glucose (mmol/L)' },
  { code: 'EMERGENCY_HYPOGLYCAEMIA',               label: 'Hypoglycaemia',                    category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_PURPURIC_RASH',               label: 'Purpuric rash',                    category: 'EMERGENCY', carriesNumeric: false },
  { code: 'EMERGENCY_BURN_FACE_INHALATION',        label: 'Burn — face / inhalation involvement', category: 'EMERGENCY', carriesNumeric: false },

  // ── Pediatric Emergency (Section 1b) ──
  { code: 'PEDS_EMERGENCY_CENTRAL_CYANOSIS',       label: 'Central cyanosis (pediatric)',     category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_PULSE_LOW_OR_ABSENT',    label: 'Pulse low or absent (pediatric)',  category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_COLD_HANDS_COMPOSITE',   label: 'Cold hands — any associated sign', category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_COLD_HANDS_LETHARGIC',   label: 'Cold hands + lethargic',           category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_COLD_HANDS_PULSE_WEAK_FAST', label: 'Cold hands + pulse weak / fast', category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_COLD_HANDS_CAP_REFILL',  label: 'Cold hands + delayed capillary refill', category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_SEVERE_DEHYDRATION',     label: 'Severe dehydration (pediatric)',   category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_DEHYDRATION_SKIN_PINCH', label: 'Dehydration — abnormal skin pinch', category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_DEHYDRATION_LETHARGY',   label: 'Dehydration + lethargy',           category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },
  { code: 'PEDS_EMERGENCY_DEHYDRATION_SUNKEN_EYES', label: 'Dehydration + sunken eyes',       category: 'PEDIATRIC_EMERGENCY', carriesNumeric: false },

  // ── mSAT Very Urgent (Section 3) ──
  { code: 'MSAT_VU_FOCAL_NEUROLOGIC_DEFICIT',      label: 'Focal neurological deficit',       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_ALTERED_MENTAL_STATUS',         label: 'Altered mental status',            category: 'MSAT_VU', carriesNumeric: true, numericLabel: 'Glucose (mmol/L)' },
  { code: 'MSAT_VU_CHEST_PAIN',                    label: 'Chest pain',                       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_POISONING_OVERDOSE',            label: 'Poisoning / overdose',             category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_PREGNANT_ABDOMINAL_PAIN',       label: 'Pregnant + abdominal pain',        category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_COUGHING_VOMITING_BLOOD',       label: 'Coughing or vomiting blood',       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_DIABETIC_HIGH_GLUCOSE',         label: 'Diabetic — high glucose',          category: 'MSAT_VU', carriesNumeric: true, numericLabel: 'Glucose (mmol/L)' },
  { code: 'MSAT_VU_AGGRESSION',                    label: 'Aggression / agitation',           category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_SHORTNESS_OF_BREATH',           label: 'Shortness of breath',              category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_BURN_OVER_20_PERCENT',          label: 'Burn over 20% body surface',       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_OPEN_FRACTURE',                 label: 'Open fracture',                    category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_THREATENED_LIMB',               label: 'Threatened limb',                  category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_EYE_INJURY',                    label: 'Eye injury',                       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_LARGE_JOINT_DISLOCATION',       label: 'Large joint dislocation',          category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_SEVERE_MECHANISM_OF_INJURY',    label: 'Severe mechanism of injury',       category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_VERY_SEVERE_PAIN',              label: 'Very severe pain',                 category: 'MSAT_VU', carriesNumeric: false },
  { code: 'MSAT_VU_PREGNANT_ABDOMINAL_TRAUMA',     label: 'Pregnant + abdominal trauma',      category: 'MSAT_VU', carriesNumeric: false },

  // ── mSAT Urgent (Section 4) ──
  { code: 'MSAT_URG_UNABLE_TO_DRINK_VOMITS',       label: 'Unable to drink / vomits',         category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_ABDOMINAL_PAIN',               label: 'Abdominal pain',                   category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_VERY_PALE',                    label: 'Very pale',                        category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_PREGNANT_VAGINAL_BLEEDING',    label: 'Pregnant + vaginal bleeding',      category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_DIABETIC_VERY_HIGH_GLUCOSE',   label: 'Diabetic — very high glucose',     category: 'MSAT_URG', carriesNumeric: true, numericLabel: 'Glucose (mmol/L)' },
  { code: 'MSAT_URG_FINGER_TOE_DISLOCATION',       label: 'Finger / toe dislocation',         category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_CLOSED_FRACTURE',              label: 'Closed fracture',                  category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_BURN_WITHOUT_URGENT_SIGNS',    label: 'Burn without urgent signs',        category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_PREGNANT_TRAUMA_NON_ABDOMINAL', label: 'Pregnant + non-abdominal trauma', category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_MODERATE_PAIN',                label: 'Moderate pain',                    category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_LACERATION_ABSCESS',           label: 'Laceration / abscess',             category: 'MSAT_URG', carriesNumeric: false },
  { code: 'MSAT_URG_FOREIGN_BODY_ASPIRATION',      label: 'Foreign body aspiration',          category: 'MSAT_URG', carriesNumeric: false },

  // ── Special considerations ──
  { code: 'SPECIAL_ACUTE_TRAUMA',                  label: 'Acute trauma',                     category: 'SPECIAL', carriesNumeric: false },
  { code: 'SPECIAL_SEIZURE_HISTORY',               label: 'Seizure history',                  category: 'SPECIAL', carriesNumeric: false },
  { code: 'SPECIAL_ASSAULT_ABUSE',                 label: 'Assault / abuse',                  category: 'SPECIAL', carriesNumeric: false },
  { code: 'SPECIAL_SUICIDE_ATTEMPT',               label: 'Suicide attempt / self-harm',      category: 'SPECIAL', carriesNumeric: false },
];

export const SIGN_BY_CODE: Record<string, ClinicalSignDefinition> = Object.fromEntries(
  CLINICAL_SIGN_DEFINITIONS.map((d) => [d.code, d]),
);

/** Human-readable category label for headings. */
export const CATEGORY_LABEL: Record<ClinicalSignCategory, string> = {
  EMERGENCY: 'Emergency Signs',
  PEDIATRIC_EMERGENCY: 'Pediatric Emergency Signs',
  MSAT_VU: 'mSAT — Very Urgent',
  MSAT_URG: 'mSAT — Urgent',
  SPECIAL: 'Special Considerations',
};

/** Order of categories on the page — emergency first, special last. */
export const CATEGORY_ORDER: ClinicalSignCategory[] = [
  'EMERGENCY',
  'PEDIATRIC_EMERGENCY',
  'MSAT_VU',
  'MSAT_URG',
  'SPECIAL',
];

/** Color tokens per category for chips, badges, banners. */
export const CATEGORY_TONE: Record<ClinicalSignCategory, { text: string; bg: string; border: string; dot: string }> = {
  EMERGENCY:           { text: 'text-red-600',     bg: 'bg-red-500/10',     border: 'border-red-500/30',     dot: 'bg-red-500' },
  PEDIATRIC_EMERGENCY: { text: 'text-rose-600',    bg: 'bg-rose-500/10',    border: 'border-rose-500/30',    dot: 'bg-rose-500' },
  MSAT_VU:             { text: 'text-orange-600',  bg: 'bg-orange-500/10',  border: 'border-orange-500/30',  dot: 'bg-orange-500' },
  MSAT_URG:            { text: 'text-yellow-600',  bg: 'bg-yellow-500/10',  border: 'border-yellow-500/30',  dot: 'bg-yellow-500' },
  SPECIAL:             { text: 'text-violet-600',  bg: 'bg-violet-500/10',  border: 'border-violet-500/30',  dot: 'bg-violet-500' },
};
